static/app/components/charts/onDemandMetricRequest.tsx

```diff
@@ -0,0 +1,126 @@
+import {doEventsRequest} from 'sentry/actionCreators/events';
+import {addErrorMessage} from 'sentry/actionCreators/indicator';
+import EventsRequest from 'sentry/components/charts/eventsRequest';
+import {t} from 'sentry/locale';
+import {EventsStats, MultiSeriesEventsStats} from 'sentry/types';
+import {DiscoverDatasets} from 'sentry/utils/discover/types';
+
+function numOfEvents(timeseriesData) {
+  return timeseriesData.data.reduce((acc, item) => {
+    const count = item[1][0].count;
+    return acc + count;
+  }, 0);
+}
+
+function applySampleRate(timeseriesData, sampleRate = 1) {
+  const scaledData = timeseriesData.data.map(([timestamp, value]) => {
+    return [timestamp, value.map(item => ({count: Math.round(item.count / sampleRate)}))];
+  });
+
+  return {
+    ...timeseriesData,
+    isExtrapolatedData: true,
+    data: scaledData,
+  };
+}
+
+class OnDemandMetricRequest extends EventsRequest {
+  fetchMetricsData = async () => {
+    const {api, ...props} = this.props;
+
+    try {
+      api.clear();
```

Comment:
do we need `api.clear()`? I don't remember needing to use it

Reply:
I used it only because `eventsRequest.tsx` does the same :D

Reply:
ok, removed it :D